### PR TITLE
Add tests for controller

### PR DESCRIPTION
**Pull request by PrangonGhose**
### Changes made in this PR
- Add rails testing to `Gemfile`
- Add tests for all controllers

### CODE REVIEW

File: spec/controllers/posts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      get '/users/:id/posts/:id'
+      expect(response).to render_template(:show)
+    end
+
+    it ' the response body includes correct placeholder text' do
+      get '/users/:id/posts'
+      expect(response.body).to include('Here is a list of posts for a given user')
+    end
+  end
+end
```
Comment (mckent05) [inline spec/controllers/posts_controller_spec.rb]: - I have an error when trying to run the tests on your project. Please make sure all tests are passing. Please fix this 🙏 
Your project:
wrknkwrvnw

File: spec/controllers/posts_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    end
+
+    it 'renders template index' do
+      get users_path
+      expect(response).to render_template(:index)
+    end
+
+    it 'renders template show' do
+      get '/users/:id/posts/:id'
+      expect(response).to render_template(:show)
```
Comment (mckent05) [inline spec/controllers/posts_controller_spec.rb:17]: - According to the project requirements,  for each action you are required to check `If the response body includes correct placeholder text`. Please check if the response for the `post_show` page renders the right placeholder

File: spec/controllers/users_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+    it 'renders template index' do
+      get users_path
+      expect(response).to render_template(:index)
+    end
+
+    it 'renders template show' do
+      get '/users/:id'
+      expect(response).to render_template(:show)
+    end
```
Comment (mckent05) [inline spec/controllers/users_controller_spec.rb:18]: - According to the project requirements, for each action you are required to check `If the response body includes correct placeholder text`. Please check if the response for the `user_show` page renders the right placeholder

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @PrangonGhose, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by PrangonGhose** [comment]
### Changes made after first review
- Added new tests to check placeholder text for `users#show` and `posts#show`

**N.B. Unfortunately I was unable to find or recreate the error from the first reviewer in `posts_controller_spec.rb` file as the error is regarding file permission to web server**
All tests are passing correctly for me
image

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @PrangonGhose, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Git Flow. :heavy_check_mark: 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**